### ISSUE TITLE
CSS Named Colors 

### ISSUE THREAD

**Issue by una** [HugoGiraudel/sass-guidelines#75]
I'm just wondering why you recommend using named colors. I always avoid them as a general rule because of the potential that they may be output as their hex value in compressed mode when referenced in maps (i.e. why you always quote named colors in maps).

I know you know this, but here's a visual reference:
https://gist.github.com/una/83cd6d4feb16b3ebcb01

So it seems like it would be a best practice to avoid them altogether for consistency.

**Comment by HugoGiraudel** [comment]
Hey Una.

Color keywords are there for a reason, and a good one: there are far simpler to deal with than hexadecimal code. Some skilled designers might be able to mentally compute a color from an hexadecimal code, but most people certainly are not.

Now, regarding your example, this is typically the piece of Sass code that should not be written. `darkolivegreen`, `firebrick`, `blanchedalmond` are colors, not strings. Because of this, they should not be interpolated in a selector. 

Along the same lines, using color names as class names is probably a bad idea for maintainability, since you are now tied to a very specific color that cannot be changed without risking any further confusion. An appropriate way to write the aforementionned snippet would be:

```scss
$colors: (
  'darkolivegreen': #556b2f,
  'firebrick': #b22222,
  'blanchedalmond': #ffebcd
);

@each $key, $value in $colors {
  .item-#{$key} {
    color: $value;
  }
}
```

Hope it's now clear. :)

**Comment by una** [comment]
Yep, I understand that these should be quoted. The human-legibility concept makes sense, I am just suggesting avoiding named colors for consistency, especially when not all of the colors are named. It should be a project-basis not individual-basis per color variable. Overall consistency > individual readability (this is a different issue)

```
$green: darkolivegreen; // ok, I could concede but still don't love the idea of setting your $green to another named color
$red: #b23223; // oh no
$light-yellow: blanchedalmond; // now its inconsistent and confusing
```

**Comment by HugoGiraudel** [comment]
That is a valid argument.

**Comment by scottzirkel** [comment]
The issue I have with using named colors period is that I'm stuck using a specific 140+ color palette. HSL/RGB/Hex gives me more flexibility as a designer. Named CSS colors have always felt like we're back to the web-safe color days.

**Comment by HugoGiraudel** [comment]
> The issue I have with using named colors period is that I'm stuck using a specific 140+ color palette. HSL/RGB/Hex gives me more flexibility as a designer. Named CSS colors have always felt like we're back to the web-safe color days.

The point is not to restrict yourself to named colors only, but to use the keyword when a color you use happens to have one.

**Comment by scottzirkel** [comment]
Sorry, I misunderstood what you were saying.

**Comment by HugoGiraudel** [comment]
Running a poll: https://twitter.com/HugoGiraudel/status/564088810733322240.

**Comment by HugoGiraudel** [comment]
You got it @una, we will abandon color keywords.

**Comment by una** [comment]
:clap::clap::clap:

**Comment by HugoGiraudel** [comment]
Raah, I'm still thinking! :x

**Comment by una** [comment]
The people have spoken!

**Comment by HoBi** [comment]
Well, the value of CSS color keywords depend of what?

If it is an standart which can be changed, why not, but if it depends on some browser, any funny browser developer can change the value, which is bad. :smile_cat: 

Btw. I rather use hexadecimal, it is cool and I can +- calculate the output color.

**Comment by HugoGiraudel** [comment]
> If it is an standart which can be changed

It is part of the CSS standard: http://www.w3.org/TR/css3-color/.

**Comment by HoBi** [comment]
@HugoGiraudel @una if you take a loot to the table of colors in section [4.3. Extended color keywords](http://www.w3.org/TR/css3-color/#svg-color), there are a lot of keywords I never learn. 

For me, it is much harder and annoing to use and even read a code with color keywords, because I am a man and I can recognize red, green, blue, yellow, brown... :joy_cat: 

You, @una, are a woman and women have, supposedly, bigger color database than men, that's why you may not have a trouble with use it.

So I vote for hexadecimal code, it is really simple to understand and I do not have to take a care that this is not 'red' BUT it is 'crimson' or 'deeppink' or... as you women do. :laughing:

**Comment by una** [comment]
I think you might be missing the point that this issue is based on. Though I don't believe that gender is in any way a valid reason for this decision, I was arguing against named colors (which would also invalidate your theory on gender preferences).

**Comment by HoBi** [comment]
@una Oh, I can see that now... :yum: Next time I will read and translate twice. :smiley_cat:

**Comment by daneden** [comment]
Additional and dumb argument pro-hex colors: hex is i18n friendly.

🌐

/me *drops mic*

**Comment by lozandier** [comment]
@daneden Hi, for educational reasons, can you enlighten me & other curious readers of this thread how is Hex more 118n friendly than other color models discussed such as HSL? & HWB?

**Comment by thebabydino** [comment]
@lozandier Well, before watching a few talks on the topic, I had never heard of [Chartreuse](https://en.wikipedia.org/wiki/Chartreuse_%28liqueur%29), [Navajo](https://en.wikipedia.org/wiki/Navajo_people), [Bisque](https://en.wikipedia.org/wiki/Bisque_%28food%29) or [honeydew](https://en.wikipedia.org/wiki/Honeydew_%28melon%29) (that's literally called "yellow melon" in my language and that's probably how I would have translated into English before knowing there's an actual word for it). Anyway, my point is those words wouldn't have given me *any* indication as to what `chartreuse` or `navajowhite` or `bisque` or `honeydew` look like.

**Comment by lozandier** [comment]
@thebabydino You're not alone there; I've had similar enlightening moments with  the use of CSS colors during my lifetime. 

I was actually appreciative looking them up in Fifth grade & through middle school somewhat when I tried a good amount of them using Netscape Compose & Microsoft's Frontpage applications back in the day.

**Comment by artemist** [comment]
To @una's point: I'd second it and say that named colors aren't always semantic- darkgray is actually lighter than gray, and there are a few other gotchas. I think a lot of web devs, myself included, have enough understanding of hex values to actually interpret the colors based on number and letter placement and value- if not perfectly, we can at least make a good guess.

**Comment by lozandier** [comment]
@artemist Devs "good guessing" RGB values has always been fascinating to me since cylindrical-coordinate color system such as `HSL` & `HWB` were heavily inspired to be created to create color models that actually were significantly more sensible to humans & how they actually perceive colors through their eyes.

**Comment by HugoGiraudel** [comment]
@HoBi Without wanting to refrain anyone's opinion here, I'd like to avoid any gender consideration on such a matter. I find it quite irrelevant and prone to trigger conflicts. :)

Anyway, you folks all made your point. We'll be moving away from CSS color keywords.

**Comment by HugoGiraudel** [comment]
Done in https://github.com/HugoGiraudel/sass-guidelines/commit/040f7bacf02ecf0a7083a00472512ff4b85739f5. Will be released in 1.2.0.
